src/generic/grid.cpp

```diff
@@ -6831,13 +6831,16 @@ void wxGrid::DrawTextRectangle(wxDC& dc,
 {
     attr.GetNonDefaultAlignment(&hAlign, &vAlign);
 
+    // The text will be displaced in final wxGrid::DrawTextRectangle function.
+    const int textMargin = 2;
```

Comment:
I just don't understand where does exactly this `2` come from... I see some `+1`s in `wxGrid::DrawTextRectangle()`, so I could understand using `1` here, but why `2`?

Is this somehow related to the cell border width?

Reply:
We are using width, not offset from the left side, so I thing it should be `1` from both sides (and it looks better).

Reply:
Let's add some `GRID_TEXT_MARGIN=1` constant then and use it in both `DrawTextRectangle()` and here then, to make it more obvious that they're the same thing.

Out of curiosity, I'd also consider changing this constant to something much bigger, e.g. 5, and checking if the display really changes consistently -- maybe some 1s or 2s hardcoded in other places need to be changed too.